### PR TITLE
Not only add, but also remove files when adopting changes in SyncMediaHandler.uploadChanges()

### PR DESCRIPTION
**Pull request by cdpm**
Added logic to not only add, but also remove files when adopting changes to media files in SyncMediaHandler.uploadChanges(), and to count deletions when updating the media usn. Also refactored SyncMediaHandler.uploadChanges().

### CODE REVIEW

**Comment by Akendo** [comment]
+1
Thank you for the swiftly response. Was able to upload the new 'changes'. Didn't verify the entry deck about this, but I'll give it later a shot. 

So far it seems good to me. Thank you very much! 

Best regards
akendo

File: AnkiServer/apps/sync_app.py [hunk 1]
```diff
@@ -100,63 +103,136 @@ def begin(self, skey):
...
+                                      "SET csum = NULL " +
+                                      " WHERE fname = ?",
+                                      [(f, ) for f in filenames])
+
+        # Remove the files from our media directory if it is present.       print('Removing %d files from media dir.' % len(filenames))
+        for filename in filenames:
+            try:
+                os.remove(os.path.join(self.col.media.dir(), filename))
+            except OSError as err:
+                print("Error when removing file '%s' from media dir: %s" %
```
Comment (dsnopek) [inline AnkiServer/apps/sync_app.py]: Why is this print()ing rather than logging this?

**Comment by dsnopek** [comment]
Wow, these changes are really extension. Since I've been committing stuff without test it (yet), I'll probably just commit this one too :-) once I have an answer about the print() vs logging.

**Comment by cdpm** [comment]
Hi, thanks. I replaced the print statements with logging calls. Logging for sync app wasn't set up on my system, so I put prints in but forgot to remove them.

**Comment by dsnopek** [comment]
Thanks!

**Pull request merged**